snippets/card-product.liquid

```diff
@@ -336,21 +336,35 @@
     </div>
   </div>
 {%- else -%}
+  {%- liquid
+    assign ratio = 1
+    if media_aspect_ratio == 'portrait'
+      assign ratio = 0.8
+    elsif media_aspect_ratio == 'adapt'
+      assign ratio = card_product.featured_media.aspect_ratio
+    endif
+    if ratio == 0 or ratio == null
+      assign ratio = 1
+    endif
+  -%}
   <div class="card-wrapper product-card-wrapper underline-links-hover">
     <div
       class="
         card card--{{ settings.card_style }}
         {% if extend_height %} card--extend-height{% endif %}
+        {% if image_shape and image_shape != 'default' %} card--shape{% endif %}
         {% if settings.card_style == 'card' %} color-{{ settings.card_color_scheme }} gradient{% endif %}
       "
-      style="--ratio-percent: 100%;"
+      style="--ratio-percent: {{ 1 | divided_by: ratio | times: 100 }}%;"
     >
       <div
         class="card__inner{% if settings.card_style == 'standard' %} color-{{ settings.card_color_scheme }} gradient{% endif %} ratio"
-        style="--ratio-percent: 100%;"
+        style="--ratio-percent: {{ 1 | divided_by: ratio | times: 100 }}%;"
       >
-        <div class="card__media">
-          <div class="media media--transparent">
+        <div class="card__media {% if image_shape and image_shape != 'default' %} shape--{{ image_shape }} color-{{ settings.card_color_scheme }} gradient{% endif %}"
+        style="{% if  image_shape == 'default' and media_aspect_ratio == 'portrait' %} top: 3rem;{% endif %}">
```

Comment:
I'm confused about why/when this `top: 3rem;` is needed 🤔 
Was there a specific scenario in which it was necessary?
Here I have a different color scheme for product cards from the theme settings and it looks like this: 
![](https://screenshot.click/09-36-nhzfp-mbuac.png)

Reply:
Originally I added the top:3rem when you could change the image ratio but since I am removing the ability to change the image ratio, the top: 3rem is no longer relevant and has been removed